### PR TITLE
dwarf-fortress: Add option to install dfhack.

### PR DESCRIPTION
**Pull request by robbinch**
This adds option to install DFHack, a Dwarf Fortress memory access library, distributed with scripts and plugins implementing a wide variety of useful functions and tools. To use, create a game directory with dfhack_install_dir and run ./dfhack in it.

### CODE REVIEW

**Comment by pSub** [comment]
cc @the-kenny @roconnor

File: pkgs/games/dwarf-fortress/default.nix [hunk 1]
```diff
@@ -18,6 +21,37 @@ let
...
+    src = fetchgit {
+      url = "https://github.com/DFHack/dfhack.git";
+      rev = "0849099f2083e100cae6f64940b4eff4c28ce2eb";
+      sha256 = "0lnqrayi8hwfivkrxb7fw8lb6v95i04pskny1px7084n7nzvyv8b";
+    };
+  };
+  dfhack_installPhase = ''
+    cat > $out/bin/dfhack_install_dir <<EOF
+    #!/bin/sh
+    test -z "\$1" && echo "This creates a Dwarf Fortress/DFHack game directory. Please specify a directory (preferably empty or non-existent) for this." && exit 1
```
Comment (the-kenny) [inline pkgs/games/dwarf-fortress/default.nix]: Why do we need to create a different directory outside the nix-store for dfhack?

**Comment by robbinch** [comment]
DFHack is a separate derivation so as not to be too invasive to the original dwarf fortress derivation. If you prefer, I can make changes to the phases inside dwarf fortress itself to support dfhack.

**Comment by the-kenny** [comment]
No - the question is why this wrapping is done inside the dwarf-fortress derivation itself and not outside with a separate wrapper?

**Comment by robbinch** [comment]
Sorry if I don't understand the question fully. If you mean why a game directory needs to be created to run dfhack, I thought this allows for more flexibility as gamers can choose to run dfhack or plain vanilla dwarf fortress. dfhack is not official software and is known to crash inexplicably at times.

Or do you mean if dfhack should be a separate package entirely? This can be done too.

**Comment by the-kenny** [comment]
The latter. For one, It's always best to keep derivations as simple as possible. It's also easier to manage.

**Comment by robbinch** [comment]
Cool, have created dfhack package.

**Comment by robbinch** [comment]
Closed and reopen as New Package PR.

**Pull request closed without merging**